### ISSUE TITLE
Improve assets precompilation 

### ISSUE THREAD

**Issue by olivierobert** [nimblehq/rails-templates#326]
## Why

The template has a workaround to precompile assets for the `production` env. Since Rails seems to require all env variable to be set, the following add a dummy value:

https://github.com/nimblehq/rails-templates/blob/0e40c04eaee851257a4c19478a15337e1ba04086/bin/docker-assets-precompile#L15-L20

However, with the new build process with `esbuild`, it seems that just setting the env variable `ASSET_ENV=production` is sufficient as per [this article](https://blog.arkency.com/dont-waste-your-time-on-assets-compilation-on-heroku/#bin_rails_assets_precompile_in_a_modern_way). So it might be time to revisit.

Also, it would be good to sync assets files to S3 as well instead of serving assets from Heroku. 
 
## Who Benefits?

Web developers who work on Ruby on Rails projects

**Comment by malparty** [comment]
@hoangmirs @andyduong1920 anyone willing to work on this one? :D 
If this one was implemented, we would not have spent time debugging the `missing adapter` issue (from #191 #264 )

Thanks!! 🙇

**Comment by andyduong1920** [comment]
Hi @malparty, I'm not sure this is related to the https://github.com/nimblehq/rails-templates/issues/264 issue :D 

I didn't catch up with the Rails assets tooling for a while; before starting this improvement, are we using esbuild as the default tool for modification? 

Also, will we target to sync the compiled assets to S3 on the same issue 🤔 should we separate that into two different topics 🤔

**Comment by malparty** [comment]
@andyduong1920 
These are 2 different issues indeed— they can be done independently I think 🤔 (starting with the assets pipeline improvement first, then configuring S3 later)

**Comment by andyduong1920** [comment]
Also @malparty FYI we might improve the `asset pipeline` but for the rake task like `bin/rails i18n:js:export` it's still required all ENV 😂 because that is the Rails way FYI

**Comment by malparty** [comment]
Yep, maybe we should use the NPM `i18n` package for running the export command... It might be more performance as well 😇 but this is at risk to duplicate the configuration for exporting.
